src/schemas/hyvaksymisEsitysSchema.tsx

```diff
@@ -0,0 +1,76 @@
+import { ValidationMode, ValidationModeState } from "common/ProjektiValidationContext";
+import * as Yup from "yup";
+import { getAineistotNewSchema, getLadatutTiedostotNewSchema } from "./common";
+import { paivamaara } from "./paivamaaraSchema";
+
+const getKunnallinenLadattuTiedostoSchema = () =>
+  Yup.object().shape({
+    tiedosto: Yup.string().nullable(),
+    nimi: Yup.string().required(),
+    jarjestys: Yup.number().integer().notRequired(),
+    uuid: Yup.string().required(),
+    kunta: Yup.number().integer().defined(),
+  });
+
+const getKunnallinenLadatutTiedostotSchema = () => Yup.array().of(getKunnallinenLadattuTiedostoSchema()).nullable();
+
+const isValidationModePublish = (validationMode: ValidationModeState) => validationMode?.current === ValidationMode.PUBLISH;
+// Scheman kentät on oletuksena non nullableja ja defined-tarkistus estää ettei kentän arvot voi olla määrittämättä (undefined)
+// defined-kenttien validointiviestit ei pitäisi tulla käyttöliittymälle koskaan sillä kenttien arvoja ei pitäisi saada määrittämättömäksi
+export const hyvaksymisEsitysSchema = Yup.object().shape({
+  oid: Yup.string().required(),
+  versio: Yup.number().integer().required(),
+  muokattavaHyvaksymisEsitys: Yup.object().shape({
+    poistumisPaiva: paivamaara({ preventFuture: false, preventPast: false })
+      .defined()
+      .when("$validationMode", {
+        is: isValidationModePublish,
+        then: (schema) => schema.required("Päivämäärä on pakollinen"),
+      }),
```

Comment:
BE:ssä tarkistetaan hyväksyttäväksi lähettämisen yhteydessä, että poistumisPaiva ei ole menneisyydessä. FE-validaation pitäisi olla yhtenäinen BE-validaation kanssa.